internal/provider/data_source_gitlab_project_milestone_test.go

```diff
@@ -0,0 +1,39 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataSourceGitlabProjectMilestone_basic(t *testing.T) {
+	testAccCheck(t)
+
+	testProject := testAccCreateProject(t)
+	testMilestone := testAccAddProjectMilestones(t, testProject, 1)[0]
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataGitlabProjectMilestoneConfig(testProject.ID, testMilestone.ID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.gitlab_project_milestone.this", "milestone_id", fmt.Sprintf("%v", testMilestone.ID)),
+					resource.TestCheckResourceAttr("data.gitlab_project_milestone.this", "title", testMilestone.Title),
+					resource.TestCheckResourceAttr("data.gitlab_project_milestone.this", "description", testMilestone.Description),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataGitlabProjectMilestoneConfig(projectID int, milestoneID int) string {
```

Comment:
Move this inline to adhere to the upcoming testing guardrails: https://github.com/gitlabhq/terraform-provider-gitlab/issues/1008